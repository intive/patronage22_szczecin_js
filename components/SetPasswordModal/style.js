import styled from 'styled-components'
import Input from '../Input/Input'

export const StyledInput = styled(Input)`
  width: 100%;

  @media (min-width: 576px) {
    width: 448px;
  }
`
