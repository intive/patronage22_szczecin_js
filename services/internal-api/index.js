import axios from 'axios'

const axiosInstance = axios.create({
  baseURL: `https://${process.env.NEXT_PUBLIC_VERCEL_URL}` || 'http://localhost:3000',
  responseType: 'json',
  withCredentials: true
})

export const getBoards = async (...args) => {
  return axiosInstance.get('/api/boards', { params: Object.assign({}, ...args) })
}
