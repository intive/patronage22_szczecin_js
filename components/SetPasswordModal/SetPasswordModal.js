import Modal from '../Modal/Modal'
import BoardsContext from '../../store/boards-context'
import { useTranslation } from 'next-i18next'
import { StyledInput } from './style'
import { useState, useContext, useEffect } from 'react'
import { updateBoard } from '../../services/internal-api'
import { ModalProvider } from '../../store/modal-context.js'

export default function SetPasswordModal ({ boardId, handleClose, isModalOpen }) {
  const { t } = useTranslation(['modals'])

  const [password, setPassword] = useState('')

  const boardsCtx = useContext(BoardsContext)

  const refreshOnSuccess = data => data.status === 204 && boardsCtx.reload()
  const updateBoardHandler = async (id, options) => refreshOnSuccess(await updateBoard(id, options))

  const handleOnChange = (event) => {
    const onlyDigitsRegex = /^[0-9\b]+$/
    if (event.target.value === '' || onlyDigitsRegex.test(event.target.value)) {
      setPassword(event.target.value)
    }
  }

  const handleSaveClick = () => {
    updateBoardHandler(boardId, { password: password })
  }

  useEffect(() => {
    if (!isModalOpen) {
      setPassword('')
    }
  }, [isModalOpen])

  const modalOptions = {
    isModalOpen,
    icon: 'lock_outline',
    handleClose,
    handleSaveClick,
    title: t('setPasswordModal.title'),
    subtitle: t('setPasswordModal.subtitle'),
    disabled: password.length < 8
  }

  return (
    <ModalProvider value={modalOptions}>
      <Modal>
        <StyledInput value={password} onChange={handleOnChange} type='password' maxLength='8' placeholder='Enter password' />
      </Modal>
    </ModalProvider>
  )
}
