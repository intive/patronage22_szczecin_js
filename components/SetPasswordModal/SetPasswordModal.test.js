/* globals describe, it, expect, jest */

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import SetPasswordModal from './SetPasswordModal'
import * as InternalApi from '../../services/internal-api'

describe('SetPasswordModal', () => {
  const mockUpdateBoard = jest.spyOn(InternalApi, 'updateBoard')

  it('should render SetPasswordModal correctly', () => {
    const { baseElement } = render(<SetPasswordModal isModalOpen='true' />)

    expect(baseElement).toMatchSnapshot()
  })

  it('should call patch option when save button is clicked', () => {
    render(<SetPasswordModal boardId='mockedId' isModalOpen />)
    expect(screen.getByRole('button', { name: /buttons.save/i })).toBeInTheDocument()
    userEvent.type(screen.getByPlaceholderText('Enter password'), '12345678')
    expect(screen.getByPlaceholderText('Enter password')).toHaveValue('12345678')
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeDisabled()
    userEvent.click(screen.getByRole('button', { name: /buttons.save/i }))
    expect(mockUpdateBoard).toHaveBeenCalledTimes(1)
    expect(mockUpdateBoard).toHaveBeenCalledWith('mockedId', { password: '12345678' })
  })
})
