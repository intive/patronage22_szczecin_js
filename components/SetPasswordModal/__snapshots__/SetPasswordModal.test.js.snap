// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetPasswordModal should render SetPasswordModal correctly 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    id="modal-portal"
  >
    <div
      class="sc-dkPtRN hBZupT"
      data-testid="backdrop"
    />
    <div
      aria-hidden="false"
      class="sc-hKwDye loeQtZ"
      role="dialog"
    >
      <div
        class="sc-eCImPb jdRPcx"
      >
        <span
          class="material-icons sc-jRQBWg jPDVsh"
        >
          lock_outline
        </span>
        <p
          class="sc-gKclnd eVXqrc"
          id="modalTitle"
        >
          setPasswordModal.title
        </p>
        <p
          class="sc-iCfMLu hKohOx"
          id="modalSubtitle"
        >
          setPasswordModal.subtitle
        </p>
      </div>
      <div
        class="sc-furwcr flDNAI"
      >
        <input
          class="sc-iqseJM hgSgod sc-egiyK cJcWzW"
          type="password"
        />
      </div>
      <div
        class="sc-pVTFL hIvRrS"
      >
        <button
          class="sc-bdvvtL jVCSJw sc-jrQzAO bsfgRs"
        >
          buttons.cancel
        </button>
        <button
          class="sc-bdvvtL dRBcfm sc-kDTinF gcMRAr"
        >
          buttons.continue
        </button>
      </div>
    </div>
  </div>
</body>
`;
